src/dispatch/plugins/dispatch_slack/messaging.py

```diff
@@ -185,3 +190,144 @@ def slack_preview(message, block=None):
+                    "placeholder": {
+                        "type": "plain_text",
+                        "text": "A summary of what you know so far. It's all right if this is incomplete."
+                    },
+                    "multiline": True
+                },
+                "label": {
+                    "type": "plain_text",
+                    "text": "Description",
+                    "emoji": True
+                }
+            },
+            {
+                "type": "input",
+                "block_id": IncidentSlackViewBlockId.type,
+                "element": {
+                    "type": "static_select",
+                    "placeholder": {
+                        "type": "plain_text",
+                        "text": "Select Incident Type",
+                        "emoji": True
+                    },
+                    "options": []
+                },
+                "label": {
+                    "type": "plain_text",
+                    "text": "Type",
+                    "emoji": True
+                }
+            },
+            {
+                "type": "input",
+                "block_id": IncidentSlackViewBlockId.priority,
+                "element": {
+                    "type": "static_select",
+                    "placeholder": {
+                        "type": "plain_text",
+                        "text": "Select Incident Priority",
+                        "emoji": True
+                    },
+                    "options": []
+                },
+                "label": {
+                    "type": "plain_text",
+                    "text": "Priority",
+                    "emoji": True
+                }
+            }
+        ]
+    }
+    slack_options_template = {
+        "text": {
+            "type": "plain_text",
+            "text": "",
+            "emoji": True
+        },
+        "value": ""
+    }
+
+    # below fields for incident type and priority are the same
+    # (label and value) are set from the caller function create_incident_open_modal
+    # if the value needs to be changed in the future to ID (from name to id) then modify them in the caller function
+
+    for incident_type in incident_types:
+        option = slack_options_template.copy()
+
+        option['text']['text'] = incident_type.get('label')
+        option['value'] = incident_type.get('value')
+
+        # the modal_view_template['blocks'][3] is the 4th block which is the incident Type
+        modal_view_template['blocks'][3]['element']['options'].append(option)
+
+    for incident_priority in incident_priorities:
+        option = slack_options_template.copy()
+
+        option['text']['text'] = incident_priority.get('label')
+        option['value'] = incident_priority.get('value')
+
+        # the modal_view_template['blocks'][4] is the 5th block which is the incident Priority
+        modal_view_template['blocks'][4]['element']['options'].append(option)
```

Comment:
Why not doing this before building the `modal_view_template`, so you can save the results in, let's say, `incident_type_options` and `incident_priority_options`, and then reference that in `options` rather than hardcoding the position?

Reply:
at a some point we have to hardcode the position of each row, i can create an another approach to template it further but, i'll have to reference to it for the row orders

Reply:
Hmm... I may be missing something here, but can you not rename `option` to `incident_type_options` in line 312 and then replace `[]` in line 270 with `incident_type_options` after looping through `incident_types`? Same for `incident_priorities`.

Reply:
hadn't thought of that approach .. nice one .. thanks for it mate